VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,130 @@
+package org.edx.mobile.discussions;
+
+import android.text.SpannableString;
+import android.text.TextUtils;
+import android.text.style.ClickableSpan;
+import android.widget.TextView;
+
+import org.edx.mobile.discussion.DiscussionTextUtils;
+import org.edx.mobile.discussion.IAuthorData;
+import org.edx.mobile.test.BaseTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Date;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertTrue;
+
+/**
+ * Created by miankhalid on 25/04/2016.
+ */
+public class DiscussionTextUtilsTest extends BaseTestCase {
+
+    private TextView dummyTextView;
+    private Runnable dummyClickListener;
+
+    @Before
+    public void initValues() {
+        dummyTextView = new TextView(context);
+        dummyClickListener = new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Author clicked");
+            }
+        };
+    }
+
+    @Test
+    public void test_DuplicatedWhitespaceRemoved() {
+        String input = "  start  end  ";
+        String expected = " start end ";
+        String output = DiscussionTextUtils.removeDuplicateWhitespaces(input).toString();
+        assertEquals(expected, output);
+    }
+
+    @Test
+    public void test_TextInTextViewNotNull() {
+        DiscussionTextUtils.setAuthorAttributionText(dummyTextView,
+                DiscussionTextUtils.AuthorAttributionLabel.POST,
+                new StubAuthorData("testAuthor", "testLabel", new Date()),
+                dummyClickListener);
+        assertNotNull(dummyTextView.getText());
+    }
+
+    @Test
+    public void test_AuthorIsTappable() {
+        DiscussionTextUtils.setAuthorAttributionText(dummyTextView,
+                DiscussionTextUtils.AuthorAttributionLabel.POST,
+                new StubAuthorData("testAuthor", "testLabel", new Date()),
+                dummyClickListener);
+        SpannableString cs = (SpannableString) dummyTextView.getText();
+        ClickableSpan[] spans = cs.getSpans(0, cs.length(), ClickableSpan.class);
+        if (config.isUserProfilesEnabled()) {
+            assertTrue(spans.length > 0);
+        } else {
+            assertTrue(spans.length == 0);
```

Comment:
Instead of merely ensuring that the `ClickableSpan` has been set, we could also test it by providing a mock `Runnable` click listener generated by Mockito.